app/Filament/Resources/ServerResource.php

```diff
@@ -24,22 +33,68 @@
 use Filament\Tables\Table;
 use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\SoftDeletingScope;
-use phpDocumentor\Reflection\Types\Self_;
+
 
 class ServerResource extends Resource
 {
     public const ICON = 'tabler-server-2';
 
     protected static ?string $model = Server::class;
-
     protected static ?string $navigationIcon = self::ICON;
+    protected static ?string $navigationLabel = 'Runners';
 
+    protected static ?string $modelLabel = 'Runners';
     protected static ?int $navigationSort = 30;
 
     public static function form(Form $form): Form
     {
         return $form
-            ->schema(Server::getForm());
+            ->schema([
+                FormSection::make('Server Information')
+                    ->columns(2)
+                    ->icon(ServerResource::ICON)
+                    ->schema([
+                        Select::make('server_type')
+                            ->options(ServerType::class)
+                            ->columnSpanFull()
+                            ->required()
+                            ->reactive(),
+                        TextInput::make('name')
+                            ->required()
+                            ->columnSpanFull()
+                            ->maxLength(255),
+                        TextInput::make('hostname')
+                            ->required(fn(Get $get) => $get('server_type') !== ServerType::LOCAL->value)
+                            ->hidden(fn(Get $get) => $get('server_type') === ServerType::LOCAL->value)
+                            ->maxLength(255),
+                        TextInput::make('ssh_port')
+                            ->default(22)
+                            ->required(fn(Get $get) => $get('server_type') !== ServerType::LOCAL->value)
+                            ->hidden(fn(Get $get) => $get('server_type') === ServerType::LOCAL->value)
+                            ->numeric(),
+                        Select::make('server_credentials_id')
+                            ->relationship('serverCredentials', 'title')
+                            ->searchable()
+                            ->required()
+                            ->columnSpanFull()
+                            ->required(fn(Get $get) => $get('server_type') !== ServerType::LOCAL->value)
+                            ->hidden(fn(Get $get) => $get('server_type') === ServerType::LOCAL->value)
+                            ->label('Select Credential')
+                            ->reactive(),
+                    ])
+                    ->headerActions([
+                        Action::make('test_connection')
+                            ->color('danger')
+                            ->label('Test Connection')
+                            ->action(function ($state) {
+                                self::testConnection($state);
+                            })
```

Comment:
i think this action should ask which credential we want to use to test if there are multiple credentials.
if there is only a single credential, we can bypass that requirement.
if there are no credentials, this should indicate that testing is not possible until the user adds a credential.